.github/workflows/ci/pkg/environment/environment.go

```diff
@@ -111,22 +118,82 @@ func New(c Config) (*PullRequestEnvironment, error) {
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
+	// Check if the pull request has changes to the `docs/` directory as that will effect who
+	// the required reviewers are.
+	docChanges, codeChanges, err := hasChanges(c.Context, pr, c.Client)
+	if err != nil {
+		// Log the error, don't fail the whole run because it couldn't detect if the pull request has docs
+		log.Errorf("error while detecting pull request for docs changes: %v, skipping assigning docs reviewers", err)
+	}
+
 	return &PullRequestEnvironment{
 		Client:           c.Client,
 		reviewers:        c.Reviewers,
 		defaultReviewers: c.Reviewers[ci.AnyAuthor],
 		Metadata:         pr,
+		HasDocsChanges:   docChanges,
+		HasCodeChanges:   codeChanges,
 	}, nil
 }
 
+// hasChanges determines if the pull request has changes to `docs/`
+func hasChanges(ctx context.Context, pr *Metadata, clt *github.Client) (hasDocsChanges bool, hasCodeChanges bool, err error) {
+	files, err := getPullRequestFiles(ctx, pr, clt)
+	if err != nil {
+		return false, true, trace.Wrap(err)
+	}
+	for _, file := range files {
+		if file.Filename == nil {
+			return false, true, trace.BadParameter("pull request file name is nil")
+		}
+		if strings.HasPrefix(*file.Filename, ci.DocsPrefix) {
+			hasDocsChanges = true
+		} else {
+			hasCodeChanges = true
+		}
+
+	}
+	return hasDocsChanges, hasCodeChanges, nil
+}
+
+// getPullRequestFiles gets all the files in the pull request.
+func getPullRequestFiles(ctx context.Context, pr *Metadata, clt *github.Client) ([]*github.CommitFile, error) {
+	// Can only request 100 files per page according to the docs.
+	// https://docs.github.com/en/free-pro-team@latest/rest/reference/pulls/#list-pull-requests-files
+	opt := &github.ListOptions{PerPage: 100}
+
+	var allFiles []*github.CommitFile
+	for {
+		files, resp, err := clt.PullRequests.ListFiles(ctx, pr.RepoOwner, pr.RepoName, pr.Number, opt)
```

Comment:
You might want to consider running your "has changes" detection one page at a time. For example, if the first page has both docs and code changes, you have all the info you need - there's no need to continue loading all of the pages into memory.

That said, this is not a major optimization and we shouldn't often have PRs with many pages of changes, so if you think the code will get to messy to implement this way then it's fine to leave as-is.

Reply:
It will get a bit messy, I'll leave it as is.